scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -159,6 +194,109 @@ trait JavaModule
    */
   def runIvyDeps: T[Agg[Dep]] = Task { Agg.empty[Dep] }
 
+  def parentDep: T[Option[Dep]] = Task { None }
+
+  /**
+   * Any BOM dependencies you want to add to this Module, in the format
+   * ivy"org:name:version"
+   */
+  def bomDeps: T[Agg[Dep]] = Task { Agg.empty[Dep] }
+
+  def allBomDeps: Task[Agg[(coursier.core.Module, String)]] = Task.Anon {
+    val modVerOrMalformed =
+      (Agg(parentDep().toSeq: _*) ++ bomDeps()).map(bindDependency()).map { bomDep =>
+        val fromModVer = coursier.core.Dependency(bomDep.dep.module, bomDep.dep.version)
+          .withConfiguration(coursier.core.Configuration.defaultCompile)
+        if (fromModVer == bomDep.dep)
+          Right((bomDep.dep.module, bomDep.dep.version))
+        else
+          Left(bomDep)
+      }
+
+    val malformed = modVerOrMalformed.collect {
+      case Left(malformedBomDep) =>
+        malformedBomDep
+    }
+    if (malformed.isEmpty)
+      modVerOrMalformed.collect {
+        case Right(modVer) => modVer
+      }
+    else
+      throw new Exception(
+        "Found parent or BOM dependencies with invalid parameters:" + System.lineSeparator() +
+          malformed.map("- " + _.dep + System.lineSeparator()).mkString +
+          "Only organization, name, and version are accepted."
+      )
+  }
+
+  /**
+   * Dependency management data
+   *
+   * Versions and exclusions in dependency management override those of transitive dependencies,
+   * while they have no effect if the corresponding dependency isn't pulled during dependency
+   * resolution.
+   *
+   * For example, the following forces com.lihaoyi::os-lib to version 0.11.3, and
+   * excludes org.slf4j:slf4j-api from com.lihaoyi::cask that it forces to version 0.9.4
+   * {{{
+   *   def dependencyManagement = super.dependencyManagement() ++ Agg(
+   *     ivy"com.lihaoyi::os-lib:0.11.3",
+   *     ivy"com.lihaoyi::cask:0.9.4".exclude("org.slf4j", "slf4j-api")
+   *   )
+   * }}}
+   */
+  def dependencyManagement: T[Agg[Dep]] = Task { Agg.empty[Dep] }
+
+  /**
+   * Data from dependencyManagement, converted to a type ready to be passed to coursier
+   * for dependency resolution
+   */
+  def dependencyManagementDict: Task[Seq[(DependencyManagement.Key, DependencyManagement.Values)]] =
```

Comment:
This looks a bit like utility task we could move to `CoursierSupport` (might not work as-is due to uses of `Dep`) or `CoursierModule`.

Reply:
That method isn't around any more (more or less inlined elsewhere)

Reply:
Actually, it became a private method (not a task anymore)